test/libsolidity/semanticTests/multiSource/alias_shadows_alias.sol

```diff
@@ -0,0 +1,21 @@
+==== Source: s1.sol ====
+function f() pure returns (uint) { return 1337; }
+function g() pure returns (uint) { return 42; }
+==== Source: s2.sol ====
+import {f as g} from "s1.sol";
+==== Source: s3.sol ====
+// imports f()->1337 as g()
+import "s2.sol";
+// imports f()->1337 as f() and
+// g()->42 as g
+import {f as f, g as g} from "s1.sol";
+contract C {
+  function foo() public pure returns (uint) {
+    // calls f()->1337 / f()->1337
+    return f() / g();
+  }
+}
+// ====
+// compileViaYul: also
+// ----
+// foo() -> 1
```

Comment:
This test should not pass (it currently passes). Rather lead to an error because of alias shadowing

Reply:
I think this might be due to de non-tree-like structure of the import graph